examples/calling-apis/chatbot/app/(langgraph)/api/langgraph/[..._path]/route.ts

```diff
@@ -1,36 +1,119 @@
-import { initApiPassthrough } from "langgraph-nextjs-api-passthrough";
+import { NextRequest, NextResponse } from "next/server";
 
 import { auth0 } from "@/lib/auth0";
 
-const getRefreshToken = async () => {
-  const session = await auth0.getSession();
-  const refreshToken = session?.tokenSet.refreshToken as string;
-  return refreshToken;
-};
-
-export const { GET, POST, PUT, PATCH, DELETE, OPTIONS, runtime } =
-  initApiPassthrough({
-    apiUrl: process.env.LANGGRAPH_API_URL,
-    apiKey: process.env.LANGSMITH_API_KEY,
-    runtime: "edge",
-    baseRoute: "langgraph/",
-    bodyParameters: async (req, body) => {
-      if (
-        req.nextUrl.pathname.endsWith("/runs/stream") &&
-        req.method === "POST"
-      ) {
-        return {
-          ...body,
-          config: {
-            configurable: {
-              _credentials: {
-                refreshToken: await getRefreshToken(),
-              },
-            },
-          },
-        };
-      }
-
-      return body;
+async function getAccessToken() {
+  const tokenResult = await auth0.getAccessToken();
+
+  if (!tokenResult?.token) {
+    throw new Error("Error retrieving access token for langgraph api.");
+  }
+
+  return tokenResult.token;
+}
+
+function addAuthContext(body: any, accessToken: string) {
+  return {
+    ...body,
+    config: {
+      configurable: {
+        _credentials: {
+          accessToken,
+        },
+      },
     },
-  });
+  };
+}
+
+async function makeLangGraphRequest(
+  endpoint: string,
+  method: string = "GET",
+  accessToken: string,
+  body?: any,
+  additionalHeaders: Record<string, string> = {}
+) {
+  const langGraphUrl = `${process.env.LANGGRAPH_API_URL}${endpoint}`;
+  const requestHeaders: Record<string, string> = {
+    "x-api-key": `Bearer ${process.env.LANGSMITH_API_KEY}`,
```

Comment:
it appears when running in the Langgraph Platform, the `x-api-key` will be required per this doc: 
https://docs.langchain.com/langgraph-platform/auth

We should probably validate this though.